libraries/architecture/src/main/kotlin/io/element/android/libraries/architecture/AsyncAction.kt

```diff
@@ -15,23 +15,27 @@ import kotlin.contracts.contract
 
 /**
  * Sealed type that allows to model an asynchronous operation triggered by the user.
+ * @param C the type of data required to confirm the operation.
+ * @param T the type of data returned by the operation.
  */
 @Stable
-sealed interface AsyncAction<out T> {
+sealed interface AsyncAction<out C, out T> {
```

Comment:
Maybe instead of adding new type everywhere, we could let the Confirming be an interface you can inherits only you have data to confirm? 


```
interface Confirming : AsyncAction<Nothing>
data object SimpleConfirmation : Confirming
```

```
val unbanUserAsyncAction: AsyncAction<Unit>
data class UnbanUserActionConfirmation(val member: RoomMember): Confirming
```

Reply:
Handled in #3667.